client/resource_group/controller/limiter.go

```diff
@@ -218,7 +224,8 @@ func (lim *Limiter) Reserve(ctx context.Context, waitDuration time.Duration, now
 	select {
 	case <-ctx.Done():
 		return &Reservation{
-			ok: false,
+			ok:  false,
+			err: ctx.Err(),
```

Comment:
Can we use `err` instead of `ok`? That would add `ErrClientResourceGroupThrottled` to Err.

Reply:
`ok` is only used for can if reserved enough tokens, but the error may include others like ctx.Error(). 

let's keep min changes in this PR first, I don't want to bring potential risks to other branches by picking.